src/auto/pixbuf.rs

```diff
@@ -0,0 +1,531 @@
+// This file was generated by gir (d8a605d) from gir-files (469db10)
+// DO NOT EDIT
+
+use Colorspace;
+use Error;
+use PixbufFormat;
+use ffi;
+#[cfg(any(feature = "v2_32", feature = "dox"))]
+use gio;
+#[cfg(any(feature = "v2_32", feature = "dox"))]
+use gio_ffi;
+use glib;
+use glib::StaticType;
+use glib::Value;
+use glib::object::Downcast;
+use glib::object::IsA;
+use glib::signal::SignalHandlerId;
+use glib::signal::connect;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use std::boxed::Box as Box_;
+use std::mem;
+use std::mem::transmute;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct Pixbuf(Object<ffi::GdkPixbuf>);
+
+    match fn {
+        get_type => || ffi::gdk_pixbuf_get_type(),
+    }
+}
+
+impl Pixbuf {
+    pub fn new(colorspace: Colorspace, has_alpha: bool, bits_per_sample: i32, width: i32, height: i32) -> Pixbuf {
+        unsafe {
+            from_glib_full(ffi::gdk_pixbuf_new(colorspace.to_glib(), has_alpha.to_glib(), bits_per_sample, width, height))
+        }
+    }
+
+    #[cfg(any(feature = "v2_32", feature = "dox"))]
+    pub fn new_from_bytes(data: &glib::Bytes, colorspace: Colorspace, has_alpha: bool, bits_per_sample: i32, width: i32, height: i32, rowstride: i32) -> Pixbuf {
+        unsafe {
+            from_glib_full(ffi::gdk_pixbuf_new_from_bytes(data.to_glib_none().0, colorspace.to_glib(), has_alpha.to_glib(), bits_per_sample, width, height, rowstride))
+        }
+    }
+
+    //pub fn new_from_data<'a, P: Into<Option<&'a /*Unimplemented*/PixbufDestroyNotify>>, Q: Into<Option</*Unimplemented*/Fundamental: Pointer>>>(data: &[u8], colorspace: Colorspace, has_alpha: bool, bits_per_sample: i32, width: i32, height: i32, rowstride: i32, destroy_fn: P, destroy_fn_data: Q) -> Pixbuf {
```

Comment:
This one could be (manually) implemented for anything that is `AsRef<u8>` or `AsMut<u8>` btw (depending on whether the pixbuf is allowed to modify the underlying data or not). See e.g. https://github.com/sdroege/gstreamer-rs/blob/963557b79f5d3a91db944ed953a8c9fcab78a6ba/gstreamer/src/buffer.rs#L100

Reply:
I don't like add manual new function at this stage